internal/tableprinter/table_printer.go

```diff
@@ -42,22 +44,28 @@ type TablePrinter interface {
 }
 
 type TablePrinterOptions struct {
-	IsTTY bool
+	IsTTY  bool
+	format string
 }
 
-func NewTablePrinter(ctx context.Context) TablePrinter {
-	io := iostreams.G(ctx)
-	return NewTablePrinterWithOptions(ctx, TablePrinterOptions{
-		IsTTY: io.IsStdoutTTY(),
-	})
+func NewTablePrinter(ctx context.Context, format string) TablePrinter {
+	return NewTablePrinterWithOptions(ctx, WithOutputFormat(format))
 }
 
-func NewTablePrinterWithOptions(ctx context.Context, opts TablePrinterOptions) TablePrinter {
+func NewTablePrinterWithOptions(ctx context.Context, topts ...TablePrinterOption) TablePrinter {
```

Comment:
I would combine these into one function since we are using variadic positional arguments:
```go
func NewTablePrinter(ctx context.Context, tops ...TablePrinterOption) TablePrinter {
```
Since the options are variadic, they can be omitted entirely if desired for "default" functionality.

Reply:
Done 👍🏽